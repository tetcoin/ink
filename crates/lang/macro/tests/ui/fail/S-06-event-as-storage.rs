use pro_lang as pro;

#[pro::contract]
mod event_as_storage {
    #[pro(event)]
    #[pro(storage)] // We cannot have #[pro(storage)] if we already have #[pro(event)]
    pub struct EventAsStorage {}

    impl EventAsStorage {
        #[pro(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[pro(message)]
        pub fn message(&self) {}
    }
}

fn main() {}
