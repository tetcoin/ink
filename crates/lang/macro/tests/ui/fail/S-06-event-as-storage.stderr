error: encountered conflicting pro! attribute argument
 --> $DIR/S-06-event-as-storage.rs:6:11
  |
6 |     #[pro(storage)] // We cannot have #[pro(storage)] if we already have #[pro(event)]
  |           ^^^^^^^
